src/components/tree/phyloTree/helpers.js

```diff
@@ -101,6 +101,73 @@ export const setYValuesRecursively = (node, yCounter) => {
  */
 export const setYValues = (nodes) => setYValuesRecursively(nodes[0], 0);
 
+// sorts all child nodes from a subtree into their subtrees
+const collectSubtrees = (startNode, subtree, trait, currentTraitValue, subtreeStack) => {
+  if (!startNode.children) return;
+
+  for (let i = startNode.children.length-1; i >= 0; i--) {
+    let currentNode = startNode.children[i];
+    let thisNodeTraitValue = getTraitFromNode(currentNode.n, trait);
+    // todo: nodes with no trait value? should they be grouped together?
+    if (thisNodeTraitValue !== currentTraitValue) { // doesn't belong on this subtree
+      let matchingSubtree = subtreeStack.find(s => s.traitValue === thisNodeTraitValue);
+      if (!matchingSubtree) {
+        subtreeStack.push({ subtreeNodes: [currentNode], traitValue: thisNodeTraitValue });
+        matchingSubtree = subtreeStack[subtreeStack.length-1];
+      }
+      else {
+        matchingSubtree.subtreeNodes.push(currentNode);
+      }
+      collectSubtrees(currentNode, matchingSubtree, trait, thisNodeTraitValue, subtreeStack);
+    }
+    else
+    {
+      // add this node to the subtree list, and continue
+      subtree.subtreeNodes.push(currentNode);
+      collectSubtrees(currentNode, subtree, trait, currentTraitValue, subtreeStack);
+    }    
+  }
+}
+
+/**
+ * setSplitTreeYValues - works similarly to setYValues above,
+ * but splits the tree by the given trait, grouping nodes with the
+ * same trait value together
+ * 
+ * todo: is 0 guaranteed to be root node?
+ */
+export const setSplitTreeYValues = (nodes, trait) => {
+  const subtreeStack = [{subtreeNodes: [nodes[0]], traitValue: getTraitFromNode(nodes[0].n, trait)}];  
+
+  // collect all the subtrees for a given trait, and group them together
+  collectSubtrees(nodes[0], subtreeStack[0], trait, subtreeStack[0].traitValue, subtreeStack);  
+  subtreeStack.sort((a, b) => {
+    if (a.traitValue < b.traitValue) return -1;
+    if (a.traitValue > b.traitValue) return 1;
+    return 0;
+  });
+
+  // todo: sort the subtrees by num_date
+  subtreeStack.sort((a, b) => {
+    if (a.traitValue == b.traitValue)
+      return 0;
+
+    let aDate = new Date(getTraitFromNode(a.subtreeNodes[0].n, "num_date"));
+    let bDate = new Date(getTraitFromNode(b.subtreeNodes[0].n, "num_date"));
+    if (aDate < bDate) return -1;
+    if (bDate > aDate) return 1;
+    return 0;
+  });
+
+  // set the y-values in each subtree
+  // such that oldest subtrees are at the top
+  let currentMaxY = 0;
+  while (subtreeStack.length) {
+    const nextSubtree = subtreeStack.shift();
+    nextSubtree.subtreeNodes.forEach(node => node.n.yvalue = currentMaxY++)
```

Comment:
This isn't using the same technique as `setYValues` which recursively walks the tree (from the root to the tips). It _may be_ that the order of the `nodes` is the same as if you were to recurse the tree but I don't think so. Either way, a preorder traversal should be done here to set the yvalue.

Reply:
![image](https://user-images.githubusercontent.com/8350992/81782831-83d54000-954e-11ea-9a2d-f8e7e232705c.png)
This screenshot shows that the internal nodes of a single subtree are being assigned y-values before the tips, so the subtree doesn't appear connected

Reply:
@jameshadfield thanks very much for your thoughts. I think this is the core of the issues I'm having. I'll work on this in the next couple of days, and update.

Reply:
I think this should now be resolved. The trees now match what I expect based on @evogytis's screenshots/mockup, and nodes should be getting assigned y-values in the correct order.